### ISSUE TITLE
Additional UX / Messaging for Captive Portal

### ISSUE THREAD

**Issue by lesleyjanenorton** [mozilla-mobile/mozilla-vpn-client#919]
┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/VPN-150)

**Comment by jess-cook03** [comment]
@jess-cook03 needs to loop in Chris for content.
UX size: Medium - Large depends on further context and needs definition

**Comment by jess-cook03** [comment]
@lesleyjanenorton can we add more info to this issue in order to clearly define the UX and content needed? Thank you!

**Comment by lesleyjanenorton** [comment]
Sure! So when the VPN detects a captive portal, a system notification is shown telling the user that they need to click on the notification to turn off the VPN and see the portal. If the user does this, great. If not, they end up in a potentially very confusing situation. In the app UI, there's nothing to really indicate that a captive portal has been detected and that the VPN should be turned off. It would be great if we could make this clearer to the user. The minimal UX/content needs are I think: light modifications to the "VPN is on" screen to indicate that a captive portal has been detected and that the user needs to turn the VPN off before they can connect to that portal. 

<img width="616" alt="Screen Shot 2021-05-03 at 9 48 54 AM" src="https://user-images.githubusercontent.com/22355127/116891802-c6c37400-abf4-11eb-99ed-397d7cc5bb24.png">

**Comment by jess-cook03** [comment]
@lesleyjanenorton Are we suggesting we need an inapp alert banner or notification as well as the system one?

**Comment by lesleyjanenorton** [comment]
@jess-cook03 I don't know that it would have to be an alert but just something in-app that lets users know 1.) that a captive portal has been detected and 2.) what their next steps should be.

**Comment by chrisrooneys** [comment]
@lesleyjanenorton @jess-cook03 

**System notification**
Header: _Wi-Fi access blocked_
Body: _This Wi-Fi network must show you a captive portal before giving access. Turn off VPN so you can see the portal._

In-app text @jess-cook03 to drop into design:
Body: _Turn off VPN to access Wi-Fi, then turn it back on. [Here's why.](https://support.mozilla.org/en-US/kb/no-internet-while-using-mozilla-vpn-public-wifi)_

**Comment by jess-cook03** [comment]
Update:
From design/content side we have updated copy 👆 and also have explored UI we could implement [here.](https://www.figma.com/file/csHpYWEVvVKKyBYufvXxsV/iOS-v2.3?node-id=124%3A937)
Perhaps for this release we can update the system notification copy with what @crooney provided above and then leave the in-app UI treatment/messaging for v2.4.... ?
🚩dependencies on l10n and qa for inapp treatment at this stage before code freeze

**Comment by jess-cook03** [comment]
@chrisrooneys can we get a status on this? Thank you!

**Comment by chrisrooneys** [comment]
@jess-cook03 content has been updated above and I tweaked it a bit in the [Figma file](https://www.figma.com/file/csHpYWEVvVKKyBYufvXxsV/iOS-v2.3?node-id=124%3A937).

Frame 310 is my preference for the in-app alert until captive portal does some of the switching for them!
<img width="237" alt="Screen Shot 2021-06-24 at 5 30 04 PM" src="https://user-images.githubusercontent.com/51548136/123340448-d9319e00-d511-11eb-8786-a31d1aae0bd7.png">

**Comment by jess-cook03** [comment]
@chrisrooneys thank you this is perfect! This is then marked as complete from design side for v2.4 ✅
cc @bakulf @lesleyjanenorton @strseb

**Comment by chrisrooneys** [comment]
@bakulf @lesleyjanenorton @strseb Here's the link we should use for "Here's why"

https://support.mozilla.org/en-US/kb/no-internet-while-using-mozilla-vpn-public-wifi

**Comment by jess-cook03** [comment]
@lesleyjanenorton can we confirm this change will be in for v2.5?

**Comment by strseb** [comment]
Sorry i missed this notification, i have a concern here. 
When the user hits a captive portal this means the portal is actively blocking the internet connection (including the vpn). So until this is resolved the user is essentially offline. 
So if the user in this state clicks a link to a website, nothing would happen except a "not connected to the internet error" i'm afraid. 
So we kinda need to have this "in-app", i think.

**Comment by data-sync-user** [comment]
➤ Peter Benvenuto commented:

Basti (Sebastian) Streich Lesley Norton Here’s the[ figma spec|https://www.figma.com/file/OQpyNjmowo3BIE23WGnOCM/Captive-Portal-Additional-Messaging?node-id=14%3A1429] for this issue, let me know if you had any questions.

CC Adrienne Davenport

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Verified this as fixed on the latest VPN 2.7 build on Windows 10.

The UX/messaging additional improvements are displayed while connecting to a Captive Portal network.

We found and logged a new related issue separately: https://mozilla-hub.atlassian.net/browse/VPN-1524 ( https://mozilla-hub.atlassian.net/browse/VPN-1524|smart-link ) .

We can marked this as verified-fixed.
